src/main/scala/exercise2/lib/Synthesizer.scala

```diff
@@ -0,0 +1,64 @@
+package exercise2.lib
+
+import exercise2.{QueueLike, Queue}
+
+import scala.annotation.tailrec
+import scala.util.Random
+
+object Synthesizer {
+  val ENERGY_DECAY_FACTOR = 0.996
+
+  /**
+   * Karplus-Strong Part 1
+   *
+   * Create a function def whiteNoise(frequency:Int=440, volume:Double = 1.0): Queue[Double]
+   * that returns a queue containing a total of frequency elements of random values between .5 and -.5 multiplied by volume.
+   * Frequency must be greater than zero and volume is between 0 and 1.
+   * */
+  def whiteNoise(frequency: Int = 440, volume: Double = 1.0): QueueLike[Double] = {
+    if (frequency <= 0 || volume < 0 || volume > 1) {
+      throw new IllegalArgumentException
+    }
+
+    @tailrec
+    def fillQueue(queue: QueueLike[Double], count: Int): QueueLike[Double] = {
```

Comment:
clean! Instead of using recursion, you could also achieve that by using generators and `foldLeft` with a starting value of an empty Queue.

```scala
(1 to frequency).map(_ => (Random.between(-0.5, 0.5)) * volume).foldLeft(.../*TODO*/)
```